plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/async/AsyncTransferManager.java

```diff
@@ -353,4 +364,67 @@ private void deleteUploadedObject(S3AsyncClient s3AsyncClient, UploadRequest upl
             return null;
         });
     }
+
+    /**
+     * Fetches a part of the blob from the S3 bucket and transforms it to an {@link InputStreamContainer}, which holds
+     * the stream and its related metadata.
+     * @param s3AsyncClient Async client to be utilized to fetch the object part
+     * @param bucketName Name of the S3 bucket
+     * @param blobName Identifier of the blob for which the parts will be fetched
+     * @param partNumber Part number for the blob to be retrieved
+     * @return A future of {@link InputStreamContainer} containing the stream and stream metadata.
+     */
+    @ExperimentalApi
+    public CompletableFuture<InputStreamContainer> getBlobPartInputStreamContainer(
```

Comment:
I'm not sure any of these methods belong in "AsyncTransferManager". They are wrapping the s3 client and aren't using the transfer manager's executors to do any multi threading.

Reply:
They are not currently, and all of them are `static`. I am thinking of moving them within `S3BlobContainer`. Thoughts?